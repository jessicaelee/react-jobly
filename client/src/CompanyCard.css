.Company.card {
    margin: 8px 0px;
    text-decoration: none;
    color: #494848;
    border: 2px solid transparent;
}

.Company.card:hover {
    border: 2px solid slategray;
}

img.CCard {
    width: 50px;
    height: 50px;
    vertical-align: middle;
    object-fit: cover;
}

