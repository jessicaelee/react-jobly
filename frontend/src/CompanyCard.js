import React from 'react';
import { NavLink } from 'react-router-dom';
import "./CompanyCard.css"

function CompanyCard(company) {
  const { handle, name, logo_url, description } = company.company;
  console.log(company)

  return (

    <NavLink exact to={`companies/${handle}`} className="Company card">
      <div className="CompanyCard-Body card-body">
        <h6 className="card-title d-flex justify-content-between">
          <span className="text-capitalize"> {name}
          </span>
          <img className="CCard" src={logo_url} alt={handle} />
        </h6>
        <p> {description} </p>
      </div>
    </NavLink>
  );
}

export default CompanyCard;