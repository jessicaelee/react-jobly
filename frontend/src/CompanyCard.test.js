import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  let company = {
    handle: "test_handle",
    name: "Test Company",
    description: "This is a test company.",
    logo_url:
      "https://www.washingtonpost.com/resizer/uwlkeOwC_3JqSUXeH8ZP81cHx3I=/arc-anglerfish-washpost-prod-washpost/public/HB4AT3D3IMI6TMPTWIZ74WAR54.jpg"

  };
  render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );
});

it("matches snapshot with logo", function () {
  let company = {
    handle: "test_handle",
    name: "Test Company",
    description: "This is a test company.",
    logo_url:
      "https://www.washingtonpost.com/resizer/uwlkeOwC_3JqSUXeH8ZP81cHx3I=/arc-anglerfish-washpost-prod-washpost/public/HB4AT3D3IMI6TMPTWIZ74WAR54.jpg"

  };
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot without logo", function () {
  let company = {
    handle: "test_handle",
    name: "Test Company",
    description: "Become a great dog."
  };
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard company={company} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
