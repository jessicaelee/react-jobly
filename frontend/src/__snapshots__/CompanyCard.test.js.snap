// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with logo 1`] = `
<DocumentFragment>
  <a
    class="Company card"
    href="/companies/test_handle"
  >
    <div
      class="CompanyCard-Body card-body"
    >
      <h6
        class="card-title d-flex justify-content-between"
      >
        <span
          class="text-capitalize"
        >
           Test Company
        </span>
        <img
          alt="test_handle"
          class="CCard"
          src="https://www.washingtonpost.com/resizer/uwlkeOwC_3JqSUXeH8ZP81cHx3I=/arc-anglerfish-washpost-prod-washpost/public/HB4AT3D3IMI6TMPTWIZ74WAR54.jpg"
        />
      </h6>
      <p>
         This is a test company. 
      </p>
    </div>
  </a>
</DocumentFragment>
`;

exports[`matches snapshot without logo 1`] = `
<DocumentFragment>
  <a
    class="Company card"
    href="/companies/test_handle"
  >
    <div
      class="CompanyCard-Body card-body"
    >
      <h6
        class="card-title d-flex justify-content-between"
      >
        <span
          class="text-capitalize"
        >
           Test Company
        </span>
        <img
          alt="test_handle"
          class="CCard"
        />
      </h6>
      <p>
         Become a great dog. 
      </p>
    </div>
  </a>
</DocumentFragment>
`;
